ldi-core/ldes-client-v2/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/inmemory/InMemoryTreeNodeRecordRepository.java

```diff
@@ -0,0 +1,58 @@
+package ldes.client.treenodesupplier.repository.inmemory;
+
+import ldes.client.treenodesupplier.domain.entities.TreeNodeRecord;
+import ldes.client.treenodesupplier.domain.services.TreeNodeRecordComparator;
+import ldes.client.treenodesupplier.domain.valueobject.TreeNodeStatus;
+import ldes.client.treenodesupplier.repository.TreeNodeRecordRepository;
+
+import java.util.*;
+import java.util.stream.Stream;
+
+public class InMemoryTreeNodeRecordRepository implements TreeNodeRecordRepository {
+
+	List<TreeNodeRecord> notVisited = new ArrayList<>();
```

Comment:
can these be private and final?

Reply:
private, yes
final, no, since we destroy state by re-initializing.